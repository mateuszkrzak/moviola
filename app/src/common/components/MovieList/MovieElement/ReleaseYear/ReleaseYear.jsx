import React from 'react';
import PropTypes from 'prop-types';

import './ReleaseYear.scss';

import formatYear from '../../../../utils/format-year';

export default class ReleaseYear extends React.Component {
  static defaultProps = {
    releaseDate: 'No release date provided',
  };

  static propTypes = {
    releaseDate: PropTypes.string,
  };

  render() {
    return <span styleName="title">{formatYear(this.props.releaseDate)} </span>;
  }
}
