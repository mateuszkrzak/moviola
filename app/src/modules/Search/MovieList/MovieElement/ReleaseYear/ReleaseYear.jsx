import React from 'react';
import CSSModules from 'react-css-modules';
import PropTypes from 'prop-types';

import styles from './ReleaseYear.scss';

class ReleaseYear extends React.Component {
  static defaultProps = {
    releaseDate: 'No release date provided',
  };

  formatYear = date => new Date(date).getFullYear();

  render() {
    return <span styleName="title">{this.formatYear(this.props.releaseDate)} </span>;
  }
}

ReleaseYear.propTypes = {
  releaseDate: PropTypes.string,
};

export default CSSModules(ReleaseYear, styles);
