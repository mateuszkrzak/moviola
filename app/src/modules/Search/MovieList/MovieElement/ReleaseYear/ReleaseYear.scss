.title {
  text-transform: uppercase;
  border: 1px solid black;
  display: inline;
  padding: 5px 10px;
  border-radius: 3px;
}
