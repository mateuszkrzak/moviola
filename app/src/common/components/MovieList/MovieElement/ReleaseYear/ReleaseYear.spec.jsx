import { shallow } from 'enzyme';
import React from 'react';
import ReleaseYear from './ReleaseYear';

test('ReleaseYear component renders "No release date provided" if no passed props', () => {
  const wrapper = shallow(<ReleaseYear />);
  expect(wrapper).toMatchSnapshot();
});

test('ReleaseYear component renders release year', () => {
  const mockReleaseDate = { releaseDate: '2018-01-31' };
  const wrapper = shallow(<ReleaseYear {...mockReleaseDate} />);
  expect(wrapper).toMatchSnapshot();
});
