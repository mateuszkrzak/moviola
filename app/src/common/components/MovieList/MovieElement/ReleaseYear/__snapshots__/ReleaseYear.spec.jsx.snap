// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseYear component renders "No release date provided" if no passed props 1`] = `
<span
  styleName="title"
>
  No release date provided
   
</span>
`;

exports[`ReleaseYear component renders release year 1`] = `
<span
  styleName="title"
>
  2018
   
</span>
`;
